apps/explorer/views.py

```diff
@@ -156,34 +140,67 @@ def form_invalid(self, form):
         return HttpResponseRedirect(redirect_to)
 
 
-class PixelSetDetailView(LoginRequiredMixin, DetailView):
+class PixelSetDetailView(LoginRequiredMixin, FormMixin, DetailView):
 
+    form_class = PixelSetExportPixelsForm
+    http_method_names = ['get', 'post']
     model = PixelSet
-    template_name = 'explorer/pixelset_detail.html'
     pixels_limit = 100
+    template_name = 'explorer/pixelset_detail.html'
+
+    def get_omics_units(self):
+
+        return self.request.session.get('omics_units', [])
```

Comment:
As you can store python objects in the session, what about "namespacing" stored data as follow?

```python
request.session = {
    'export': {
        'pixels': {
            'omics_units': []
        }
    }
}
```

Reply:
ah I did not do that yet.. Can I do that in a follow-up patch?

Reply:
Created an issue: https://github.com/Candihub/pixel/issues/174

Reply:
Yeah sure!